x/supply/client/rest/query.go

```diff
@@ -19,11 +28,40 @@ func RegisterRoutes(ctx context.CLIContext, r *mux.Router) {
 
 func circulatingSupplyHandler(ctx context.CLIContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		res, err := query.NewRawClient(ctx, types.ModuleName).CirculatingSupply()
-		if err != nil {
-			rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
-			return
+		q := r.URL.Query().Get("q")
+		if len(q) == 0 {
+			res, err := query.NewRawClient(ctx, types.ModuleName).CirculatingSupply()
+			if err != nil {
+				rest.WriteErrorResponse(w, http.StatusInternalServerError, err.Error())
+				return
+			}
+			rest.PostProcessResponse(w, ctx, res)
+		} else {
```

Comment:
We can avoid `else` here

```suggestion
			rest.PostProcessResponse(w, ctx, res)
			return
		}
```